tensorflow/compiler/xla/literal.cc

```diff
@@ -1225,6 +1225,7 @@ typename std::enable_if<(sizeof(NativeSrcT) != sizeof(NativeDestT)),
                         Literal>::type
 BitcastBetweenNativeTypes(const LiteralBase& src_literal) {
   LOG(FATAL) << "Invalid bitcast between types of different sizes.";
+  return {};
```

Comment:
I will add the [[noreturn]] + throw tonight or tomorrow night

Reply:
I also stumbled upon this issue in my PR #42676 and noticed that `LOG(FATAL)` internally eventually calls `abort()`, but MSVC doesn't get it in debug mode... A third option would be to explicitly invoke `std::abort()` again so that even MSVC gets it.